src/main/java/teammates/ui/newcontroller/DeleteInstructorAction.java

```diff
@@ -0,0 +1,39 @@
+package teammates.ui.newcontroller;
+
+import org.apache.http.HttpStatus;
+
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.Const;
+
+/**
+ * Action: deletes an instructor from a course.
+ */
+public class DeleteInstructorAction extends Action {
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        if (userInfo.isAdmin) {
+            return;
+        }
+        // TODO allow access to instructors with modify permission
```

Comment:
Just to take note. We need to enforce the constraint that at least one instructor should be in the course.